diag_manager/diag_output.F90

```diff
@@ -212,6 +212,7 @@ SUBROUTINE diag_output_init(file_name, FORMAT, file_title, file_unit,&
         if (.not.check_if_open(fileob)) then
                call open_check(open_file(fileobjND, trim(fname_no_tile)//".nc."//trim(mype_string), "overwrite", &
                             nc_format="64bit", is_restart=.false.))
+               call register_global_attribute(fileobjND, "NumFilesInSet", 0)
```

Comment:
This will add the attribute to any file with the `fileobjND`.  Will this add problems to files that don't have to be combined or use the X/Y axes?  

Can this be added where the axis is written?  Will it write it twice if the x-axis and y-axis are both register?  @menzel-gfdl maybe you have some insight into these questions.

Reply:
You can add a check `if (global_att_exists(fileobjND, "NumFilesinSet")) then...` to prevent potential errors from trying to register the attribute multiple times.

Reply:
If this only needs to be done when there is a x or y axis, then perhaps it should be moved and checked as @wrongkindofdoctor suggested when the[ axis is being registered](https://github.com/NOAA-GFDL/FMS/blob/master/diag_manager/diag_output.F90#L389)

Reply:
It was put here to mimic the behavior of `mpp_open`, which we know is compatible with `mppnccombine`.  It is not clear to me if/why `mppnccombine` requires this attribute for regional files, since it usually sets this value to zero even when the region is split over multiple ranks.

Reply:
I realize why this code exists.  But this will write the attribute for all of the files that aren't regional diagnostics.  I don't think we want that

Reply:
Upon further consultation and review, this ia fine.